lib/puppet/defaults.rb

```diff
@@ -1395,7 +1395,25 @@ def self.default_diffargs
     },
     :server => {
       :default => "puppet",
-      :desc => "The puppet master server to which the puppet agent should connect."
+      :desc => "The puppet master server to which the puppet agent should connect.",
+      :call_hook => :on_initialize_and_write,
+      :hook => proc { |value|
+        if Puppet.settings.set_by_config?(:server) && Puppet.settings.set_by_config?(:server_list)
+          Puppet.deprecation_warning('Attempted to set both server and server_list. Server setting will not be used.', :SERVER_DUPLICATION)
```

Comment:
Ah, this differs subtly from the route we went in pxp-agent. Because we made it command-line only, we could say that it's an error if both are in the config file, but if you use the command-line it overrides either. Here you don't differentiate between things set on the command-line or in a config file.